driver/csiplugin/nodeserver.go

```diff
@@ -39,28 +39,68 @@ type ScaleNodeServer struct {
 }
 
 const hostDir = "/host"
+const mountPath = "/mnt"
 const errStaleNFSFileHandle = "stale NFS file handle"
 
 const nodePublishMethod = "NODEPUBLISH_METHOD"
 const nodePublishMethodSymlink = "SYMLINK"
 
+const isOpenShiftCluster = "IS_OpenShift"
+const mountPathLength = 6
+
 // checkGpfsType checks if a given path is of type gpfs and
 // returns nil if it is a gpfs type, otherwise returns
 // corresponding error.
-func checkGpfsType(path string) (bool error) {
-	args := []string{"-f", "-c", "%T", path}
-	out, err := executeCmd("stat", args)
-	if err != nil {
-		return fmt.Errorf("checkGpfsType - stat [%s] failed with error [%v]", path, err)
-	}
-	outString := string(out[:])
-	outString = strings.TrimRight(outString, "\n")
-	if outString != "gpfs" {
-		return fmt.Errorf("checkGpfsType - [%s] is not a valid gpfs path. reported type is [%s]", strings.TrimPrefix(path, hostDir), outString)
-	}
-	return nil
+func checkGpfsType(ctx context.Context, path string) error {
+        gpfsPaths := getGpfsPaths(ctx)
+        isGpfsPath := false
+        for _, gpfsPath := range gpfsPaths {
+                if strings.HasPrefix(path, gpfsPath) {
+                        isGpfsPath = true
+                        break
+                }
+        }
+
+        if !isGpfsPath {
+                return fmt.Errorf("checkGpfsType: the path [%s] is not a valid gpfs path ", strings.TrimPrefix(path, hostDir))
+        }
+
+        return nil
+}
+
+func getGpfsPaths(ctx context.Context) []string {
+        var gpfsPaths []string
+        gpfsPathCmd := `cat /proc/mounts | grep -i "gpfs"`
```

Comment:
Is it necessary to do case insensitive grep or `-i` can be removed from here?

Reply:
Done